src/tasks/base.py

```diff
@@ -197,7 +188,7 @@ def losses(self):
         return self._losses
 
     @property
-    def input_shapes(self) -> List[Tuple[int, int, int]]:
+    def input_shapes(self) -> List[Tuple[int, int, int, int]]:
```

Comment:
There are might be any number of input dimensions, so I suggest to use `List[List[int]]`